src/d/actor/d_a_player_hook.inc

```diff
@@ -157,20 +255,133 @@ BOOL daPy_lk_c::procHookshotSubject() {
 
 /* 8014B518-8014B5E8       .text procHookshotMove_init__9daPy_lk_cFv */
 BOOL daPy_lk_c::procHookshotMove_init() {
-    /* Nonmatching */
+    if (mCurProc == daPyProc_HOOKSHOT_MOVE_e) {
+        return false;
+      } else {
+        commonProcInit(daPyProc_HOOKSHOT_MOVE_e);
+        if (m34B8 == 3) {
+          current.angle.y = shape_angle.y + -0x4000;
+        } else {
+          m34B8 = 2;
+          current.angle.y = shape_angle.y + 0x4000;
+        }
+        daHookshot_c *hookshot = (daHookshot_c *)mActorKeepEquip.getActor();
+        if (fopAcM_GetParam(hookshot) == 0) {
+          setActAnimeUpper(LKANM_BCK_HOOKSHOTWAIT, UPPER_MOVE2_e, 1.0f, 0.0f, -1,
+                           -1.0f);
+          setBlendAtnMoveAnime(daPy_HIO_basic_c0::m.field_0xC);
+        }
+        dComIfGp_setPlayerStatus0(0, daPyStts0_HOOKSHOT_AIM_e);
+        return true;
+      }
 }
 
 /* 8014B5E8-8014B74C       .text procHookshotMove__9daPy_lk_cFv */
 BOOL daPy_lk_c::procHookshotMove() {
-    /* Nonmatching */
+    if (!checkItemModeActorPointer()) {
+        return true;
+      }
+      daHookshot_c *hookshot = (daHookshot_c *)mActorKeepEquip.getActor();
+      mSightPacket.offDrawFlg();
+      if (fopAcM_GetParam(hookshot) == 0) {
+        if (mpAttnActorLockOn != NULL) {
+          setSpeedAndAngleAtnActor();
+        } else {
+          if (m355E == 0) {
+            if (!mpAttention->Lockon()) {
+              resetActAnimeUpper(UPPER_MOVE2_e, -1.0f);
+              goto LAB_8014b69c;
+            }
+          }
+          setSpeedAndAngleAtn();
+        }
+      }
+    LAB_8014b69c:
```

Comment:
If I'm not mistaken this goto can be resolved as if/elseif/else if the nesting is reduced
```suggestion
        } else if (m355E == 0 && !mpAttention->Lockon()) {
            resetActAnimeUpper(UPPER_MOVE2_e, -1.0f);
        } else {
            setSpeedAndAngleAtn();
        }
```